config_converter/config_converter.py

```diff
@@ -0,0 +1,215 @@
+            'read_lines_limit', 'skip_refresh_on_startup'
+        ]
+        self.supported_plugins = ['in_tail']
+        self.unsupported_plugins = ['in_forward', 'in_syslog']
+
+    def _parser_dir(self, specific: dict, p: config_pb2.Param) -> None:
+        """Create parser dir in master config."""
+        parser_type_map = {
+            'multiline': 'multiline',
+            'none': 'none',
+            'regex': 'regex',
+            'apache2': 'regex',
+            'apache_error': 'regex',
+            'json': 'json',
+            'nginx': 'regex'
+        }
+        specific['parser'] = specific.get('parser', {})
+        if p.name == 'expression':
+            specific['parser']['regex_parser_config'] = \
+                    specific['parser'].get('regex_parser_config', {})
+            specific['parser']['regex_parser_config'][p.name] = p.value
+        elif p.name == 'format' or p.name == '@type':
+            if p.value not in parser_type_map:
+                print('unknown parser format type ' + p.value)
+            else:
+                specific['parser']['type'] = parser_type_map[p.value]
+        else:
+            specific['parser']['multiline_parser_config'] = \
+                    specific['parser'].get('multiline_parser_config', {})
+            if p.name in [f'format{i}' for i in range(1, 21)]:
+                format_number = p.name[6:]
+                specific['parser']['multiline_parser_config']\
+                        [f'format_{format_number}'] = p.value
+            elif p.name == 'multiline_flush_interval':
+                specific['parser']['multiline_parser_config']['flush_interval']\
+                        = p.value
+            else:
+                specific['parser']['multiline_parser_config'][p.name] = p.value
+
+
+def read_file(path: str) -> str:
+    """Returns content of file at path."""
+    with open(path, 'rt') as f:
+        return f.read()
+
+
+def get_object(args) -> config_pb2.Directive:
+    """Run ruby exec file and get message object."""
+    try:
+        subprocess.run(
+            ['config_converter/config_parser_ruby/bin/config_parser'] + args,
+            check=True)
+    except subprocess.CalledProcessError:
+        sys.exit()
+    if not os.path.exists(args[-1] + '/config.json'):
+        sys.exit()
+    return json_format.Parse(read_file(args[-1] + '/config.json'),
+                             config_pb2.Directive())
+
+
+def write_to_yaml(result: dict, path: str, name: str) -> None:
+    """Writes created result dictionary to a yaml file."""
+    with open(f'{path}/{name}.yaml', 'w') as f:
+        yaml.dump(result, f)
+
+
+def main():
+    """Parse CLI arguments, then create converter object."""
+    config_obj: config_pb2.Directive = get_object(sys.argv[1:])
+    fluentd_path, master_path = sys.argv[1], sys.argv[-1]
+    file_name = os.path.basename(fluentd_path)
+    write_to_yaml(ConfigConverter(config_obj).result, master_path, file_name)
```

Comment:
`result` sounds like an internal variable, and Config Converter object can export it as yaml and make `write_to_yaml` as a public function.

An alternative is that we don't need a ConfigConverter Class. I think I am fine either way, I'll let Dave comment on the style.